src/core/geometry/qgsrectangle.cpp

```diff
@@ -112,16 +117,12 @@ QString QgsRectangle::asWktCoordinates() const
 
 QString QgsRectangle::asWktPolygon() const
 {
-  QString rep =
-    QLatin1String( "POLYGON((" ) +
-    qgsDoubleToString( mXmin ) + ' ' + qgsDoubleToString( mYmin ) + QLatin1String( ", " ) +
-    qgsDoubleToString( mXmax ) + ' ' + qgsDoubleToString( mYmin ) + QLatin1String( ", " ) +
-    qgsDoubleToString( mXmax ) + ' ' + qgsDoubleToString( mYmax ) + QLatin1String( ", " ) +
-    qgsDoubleToString( mXmin ) + ' ' + qgsDoubleToString( mYmax ) + QLatin1String( ", " ) +
-    qgsDoubleToString( mXmin ) + ' ' + qgsDoubleToString( mYmin ) +
-    QStringLiteral( "))" );
-
-  return rep;
+  return QStringLiteral( "POLYGON((%1 %2, %3 %2, %3 %4, %1 %4, %1 %2))" ).arg(
+           qgsDoubleToString( mXmin ),
+           qgsDoubleToString( mYmin ),
+           qgsDoubleToString( mXmax ),
+           qgsDoubleToString( mYmax )
+         );
 }
```

Comment:
Shouldn't this method also consider Null rectangles and print as POLYGON EMPTY ?

Reply:
I didn't want to overlap with your work -- I'll leave you to handle all the changes relating to null/empty